<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>
  <arg name="open_rviz" default="false"/>
  <arg name="open_gazebo" default="true"/>
  <arg name="team_size" value="2"/>
<!-- other robot map-->
<arg name="watkin_map" default="$(find multi_amr_nectec)/maps/watkin_lab.yaml"/>
<arg name="watkin_world" default="$(find multi_amr_nectec)/worlds/watkinslab.world"/>
 <arg name="map_empty" default="$(find multi_amr_nectec)/maps/map.yaml"/>
 <arg name="emptyworld" default="$(find turtlebot3_gazebo)/worlds/empty.world"/>
<arg name="neo_track1_world" default="$(find neo_simulation)/worlds/neo_track1.world"/>
<arg name="neo_track1_map" default="$(find neo_simulation)/maps/neo_track1.yaml"/>
<arg name="asphalt_plane" default="$(find multi_amr_nectec)/worlds/asphalt_plane.world"/>
<arg name="formation_test_world" default="$(find multi_amr_nectec)/worlds/formation_test.world"/>
<arg name="formation_test_map" default="$(find multi_amr_nectec)/maps/formation_test.yaml"/>
  <!-- Map server --> 
 <node pkg="map_server" type="map_server" name="map_server" args="$(arg formation_test_map)" respawn="false" >
    <param name="frame_id" value="/map" />
  </node>
  <!-- start gazebo --> 
<include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg formation_test_world)"/> 
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg open_gazebo)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
</include>
 <!-- RVIZ  --> 
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
         args="-d $(find multi_amr_nectec)/rviz/2_amr.rviz"/>
  </group>


<!-- AMR 0  -->
  <group ns="amr_0">
         <param name="tf_prefix" value="amr_0"/>
  <include file="$(find multi_amr_nectec)/launch/Gazebo_sim/single_ridgeback.launch" >
     <arg name="init_pose_gazebo" value="-x 0 -y 0 -z 0.1 -Y 1.57079632679 -P 0 -R 0" />
      <arg name="init_pose_rviz" value="0 0 0 0 0 0" />    
      <arg name="robot_name" value="amr_0" />
      <arg name="name_space" value="amr_0"/>

  </include>
  </group>
 <!--
 
  <include file="$(find multi_amr_nectec)/launch/spawn_load.launch" >
   </include>
 -->
<!-- AMR 0 
  <group ns="amr_1">
  <param name="tf_prefix" value="amr_1"/>
  <include file="$(find multi_amr_nectec)/launch/spawn_new_ridgeback.launch" >
      
      <arg name="initial_pose" value="-x 0 -y 0 -z 0 -R 0 -P 0 -Y 0"/>    
      <arg name="name_space" value="amr_1"/>

  </include>
  </group>
 -->

</launch>